activesupport/lib/active_support/core_ext/hash/conversions.rb

```diff
@@ -106,7 +106,25 @@ class << self
     #   # => {"hash"=>{"foo"=>1, "bar"=>2}}
     #
     # +DisallowedType+ is raised if the XML contains attributes with <tt>type="yaml"</tt> or
-    # <tt>type="symbol"</tt>. Use <tt>Hash.from_trusted_xml</tt> to parse this XML.
+    # <tt>type="symbol"</tt>. Use <tt>Hash.from_trusted_xml</tt> to
+    # parse this XML.
+    #
+    # Custom +disallowed_types+ can also be passed in the form of an
+    # array.
+    #
+    #   xml = <<-XML
+    #     <?xml version="1.0" encoding="UTF-8"?>
+    #       <hash>
+    #         <foo type="integer">1</foo>
+    #         <bar type="string">"David"</bar>
+    #       </hash>
+    #   XML
+    #
+    #   hash = Hash.from_xml(xml, ['integer'])
+    #   # => ActiveSupport::XMLConverter::DisallowedType: Disallowed type attribute: "integer"
+    #
+    # Note that the default disallowed types - Symbol and YAML will be
+    # ignored in case custom disallowed types are passed by user.
```

Comment:
```ruby
def initialize(xml, disallowed_types = nil)
   @xml = normalize_keys(XmlMini.parse(xml))
   @disallowed_types = disallowed_types || DISALLOWED_TYPES
end
```
If I pass `[integer]` as custom disallowed type then default DISALLOWED_TYPES i.e. Symbol and YAML will be ignored due to the `||` condition.